### DISCUSSION TITLE
Problems encountered when using the demonstrator on mobile devices

### DISCUSSION THREAD

**Discussion by Lafatalite** [pyri-project/pyri-core#18] [Q&A]
I tried using the demonstrator webpage on a mobile device, but currently, on the Jog Joint page, the buttons to increase and decrease angles do not respond when I touch them with my finger. However, when I connect a mouse to the tablet and click the buttons, they work. Additionally, when I long-press, it triggers the webpage menu. I have attempted to prevent the long-press menu by using `@contextmenu.prevent` and also tried modifying the `@mousedown` event to make the jog buttons respond to touch events. Currently, only the increase and decrease buttons on the `Joint Jog` page are not working during testing, while the `Edit Standby` and `Move to Standby` buttons are functioning properly.

**Comment by johnwason** [comment]
We have successfully used the software with a tablet interface using Ubuntu Linux, but have not tried a mobile browser. What browser are you using? We are using Firefox for our testing.

**Reply by Lafatalite** [reply]
I am using a tablet computer with Windows 11 installed. I have installed Chrome, Firefox, and Edge browsers and have tried them all, but none of them can trigger the button functionality. Additionally, long-pressing the screen triggers the right-click menu.

**Reply by Lafatalite** [reply]
I have found a solution. I have now added `touchstart` and `touchend` events to set `self.mousedown = true` and `self.mousedown = false`. The issue is that the `jog_joint` function needs to check the `self.mousedown` flag within the loop. The specific location is in the `jog_component.py` file, within the `async_jog_joints` function.

**Reply by johnwason** [reply]
I think it may be necessary to use "preventDefault()" with the touch events to stop this behavior from occurring.

https://developer.mozilla.org/en-US/docs/Web/API/Event/preventDefault

I am not sure if you will need to set `mousedown` or if it will still pass a mouse event with a touchscreen event.
